firmware/shared/src/hw/hw_i2c.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "hw_hal.h"
+#include <stdbool.h>
+
+/**
+ * @param i2c_handle The given I2C handle structure.
+ * @param target_address The device's 7 bit address value in the datasheet.
+ * @param timeout_ms Timout duration.
+ */
+typedef struct
+{
+    I2C_HandleTypeDef *i2c_handle;
+    uint16_t           target_address;
+    uint32_t           timeout_ms;
+} I2cInterface;
+
+/**
+ * @brief Check if device connected to the given I2C interface is ready for communication.
+ * @param i2c The given I2C interface.
+ * @return True if connected device is ready to communicate over I2C.
+ */
+bool hw_i2c_is_target_ready(const I2cInterface *i2c);
```

Comment:
nitpick: functions should be `camelCase`. 

So `hw_i2c_isTargetReady` (here and below).